src/Aspire.Hosting.Azure/AzureEnvironmentResource.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma warning disable ASPIRECOMPUTE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+#pragma warning disable ASPIREAZURE001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+using Aspire.Hosting.ApplicationModel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// Represents the root Azure deployment target for an Aspire application.
+/// Emits a <c>main.bicep</c> that aggregates all provisionable resources.
+/// </summary>
+public sealed class AzureEnvironmentResource : Resource
+{
+    /// <summary>
+    /// Gets or sets the Azure location that the resources will be deployed to.
+    /// </summary>
+    public object? Location { get; set; }
+
+    /// <summary>
+    /// Gets or sets the Azure resource group name that the resources will be deployed to.
+    /// </summary>
+    public object? ResourceGroupName { get; set; }
```

Comment:
We can keep these internal for now and only make them accessible via the property method?